lexicon/providers/rcodezero.py

```diff
@@ -0,0 +1,196 @@
+
+    # Delete an existing record.
+    # If record does not exist, do nothing.
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+
+        if identifier is not None:
+            rtype, name, content = self._parse_identifier(identifier)
+
+        LOGGER.debug("delete %s %s %s", rtype, name, content)
+        if rtype is None or name is None:
+            raise Exception("Must specify at least both rtype and name")
+
+        payload = self._get(
+            '/zones/{0}/rrsets?page_size=-1'.format(self.domain_id))
+
+        for rrset in payload['data']:
+            if rrset['type'] == rtype and self._fqdn_name(rrset['name']) == self._fqdn_name(name):
+                update_data = rrset
+
+                if content is None:
+                    update_data['records'] = []
+                    update_data['changetype'] = 'DELETE'
+                else:
+                    new_record_list = []
+                    for record in update_data['records']:
+                        if self._clean_content(rrset['type'], content) != record['content']:
+                            new_record_list.append(record)
+
+                    update_data['records'] = new_record_list
+                    update_data['changetype'] = 'UPDATE'
+                break
+
+        request = [update_data]
+        LOGGER.debug('request: %s', request)
+
+        self._patch('/zones/' + self.domain + '/rrsets', data=request)
+        return True
+
+    # Helpers
+    def _request(self, action='GET', url='/', data=None, query_params=None):
+        if data is None:
+            data = {}
+        if query_params is None:
+            query_params = {}
+        response = requests.request(action, self.api_endpoint + url, params=query_params,
+                                    data=json.dumps(data),
+                                    headers={
+                                        'Authorization': 'Bearer ' +
+                                                         self._get_provider_option('auth_token'),
+                                        'Content-Type': 'application/json'
+                                    })
+        # if the request fails for any reason, throw an error.
+        if response.status_code >= 400:
+            LOGGER.error('Bad Request: %s', response.text)
+        response.raise_for_status()
+        return response.json()
+
+    # generate a unique id for a give record
+    def _make_identifier(self, rtype, name, content):  # pylint: disable=no-self-use
+        return "{}/{}={}".format(rtype, name, content)
+
+    def _parse_identifier(self, identifier):  # pylint: disable=no-self-use
+        parts = identifier.split('/')
+        rtype = parts[0]
+        parts = parts[1].split('=')
+        name = parts[0]
+        content = "=".join(parts[1:])
+        return rtype, name, content
+
+    def _clean_content(self, rtype, content):
+        if rtype in ("TXT", "LOC"):
```

Comment:
There is an existing function in `BaseProvider` that can clean the entry from superfluous quotes, but it applies only to the `TXT` entries, not the `LOC`. I recognize that it is worth here an extra function for the specific usage in this provider.

Reply:
Do you have any plans to extend the function in BaseProvider?

Reply:
No, not now until another provider shows up that could benefit from it. At this moment we can think about merging the logic in `BaseProvider`. For now it is better to keep it in your provider.

Reply:
ok